libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -146,12 +154,18 @@ void CodeTransform::freeUnusedVariables(bool _popUnusedSlotsAtStackTop)
 		return;
 
 	for (auto const& identifier: m_scope->identifiers)
-		if (holds_alternative<Scope::Variable>(identifier.second))
-		{
-			Scope::Variable const& var = std::get<Scope::Variable>(identifier.second);
-			if (m_variablesScheduledForDeletion.count(&var))
-				deleteVariable(var);
-		}
+		if (Scope::Variable const* var = get_if<Scope::Variable>(&identifier.second))
+			if (m_variablesScheduledForDeletion.count(var))
+				deleteVariable(*var);
+	// Directly in a function body block, we can also delete the function arguments,
+	// which live in the virtual function scope.
+	// However, doing so after the return variables are already allocated, seems to have an adverse
+	// effect, so we only do it before that.
+	if (!returnVariablesAndFunctionExitAreSetup() && !m_scope->functionScope && m_scope->superScope && m_scope->superScope->functionScope)
```

Comment:
So yeah, as far as I can see this change alleviates the gas regression mostly, which makes some sense to me.

Reply:
What happens if there are no return variables?

Reply:
Then the condition is always false and argument slots are never reused.
Your point being that in that case it may make sense to reuse argument slots because there are no return values in the way? Not sure... not reusing them doesn't hurt in any case.